src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1048,9 +1118,15 @@ DistributedExecutionRequiresRollback(DistributedExecution *execution)
 		return IsTransactionBlock();
 	}
 
+	if (execution->useRemoteTransactionBlocks == REMOTE_TRANSACTION_BLOCKS_DISALLOWED)
+	{
+		return false;
```

Comment:
Should this be up higher among the other `return false` conditions? It's a bit weird that we could theoretically return true when it is disallowed.

Could use a comment, e.g. `/* tasks cannot run in a transaction block */`